### DISCUSSION TITLE
ModuleNotFoundError: No module named 'google'

### DISCUSSION THREAD

**Discussion by adim-att** [evilsocket/opensnitch#931] [Q&A]
**Describe the bug**
I am trying to install the deb on ubuntu 22.04, dpkg install asked for dependencies which I couldn't fix with `apt install -f`, python is installed but shows errors as seen in the screenshot, and I also tried the proposed solution which suggests `python3.10 -m pip install --ignore-installed grpcio`. I don't know what to do ..

The information is as follows :

 - OpenSnitch version : [v1.6.0-rc.5](https://github.com/evilsocket/opensnitch/releases/tag/v1.6.0-rc.5)
 - OS: Debian GNU/Linux Ubuntu 22.04
 - DE : LXQT
 - Kernel version: 5.19.0-41-generic #42~22.04.1-Ubuntu

The daemon starts fine and systemd shows active and running (Green)

The deb packages fail to install:

```
sudo dpkg -i opensnitch*
(Reading database ... 331022 files and directories currently installed.)
Preparing to unpack opensnitch_1.6.0-rc.5-1_amd64.deb ...
Unpacking opensnitch (1.6.0-rc.5-1) over (1.6.0-rc.5-1) ...
dpkg: dependency problems prevent configuration of opensnitch:
 opensnitch depends on libnetfilter-queue1; however:
  Package libnetfilter-queue1:amd64 is not configured yet.

dpkg: error processing package opensnitch (--install):
 dependency problems - leaving unconfigured
Errors were encountered while p ..
```

**Screenshots**
![ddd1e46381b](https://user-images.githubusercontent.com/85970369/236385716-f9d47624-62a6-45e9-9c7a-ccd3dc08e8d7.jpg)

**Comment by gustavo-iniguez-goya** [comment]
hi @jindiggs ,

> dpkg install asked for dependencies which I couldn't fix 

could you post the installation log of python3-opensnitch-ui? All the required dependencies are available on ubuntu 22.04, so it should install just fine.

Try installing the packages with: `$ sudo apt opensnitch* python3-opensnitch*` it'll resolve the dependencies.

On the other hand, don't run the GUI as root. Probably you ran the pip3 command as your regular user, but running the GUI as root takes the libs from others paths.

**Comment by adim-att** [comment]
> hi @jindiggs ,
> 
> > dpkg install asked for dependencies which I couldn't fix
> 
> could you post the installation log of python3-opensnitch-ui? All the required dependencies are available on ubuntu 22.04, so it should install just fine.

![screenshot-sssssc116e6de](https://user-images.githubusercontent.com/85970369/236552153-f661c715-4d36-4f55-b4a0-ef7c3e35f607.jpg)

> Try installing the packages with: `$ sudo apt opensnitch* python3-opensnitch*` it'll resolve the dependencies.

I get this : 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package opensnitch_1.6.0-rc.5-1_amd64.deb
E: Unable to locate package python3-opensnitch-ui_1.6.0-rc.5-1_all.deb
E: Unable to locate package python3-opensnitch-ui_1.6.0-rc.5-1_all.PIP.deb



> On the other hand, don't run the GUI as root. Probably you ran the pip3 command as your regular user, but running the GUI as root takes the libs from others paths.
```
Traceback (most recent call last):
  File "/usr/bin/opensnitch-ui", line 40, in <module>
    from opensnitch.service import UIService
  File "/usr/lib/python3/dist-packages/opensnitch/service.py", line 13, in <module>
    from opensnitch import ui_pb2
  File "/usr/lib/python3/dist-packages/opensnitch/ui_pb2.py", line 6, in <module>
    from google.protobuf.internal import enum_type_wrapper
ModuleNotFoundError: No module named 'google'
```

On my colleague's machine, which is Debian 11 OS, it installed with some error, but after repetition things started to work, although Ubuntu 22.04 is a little hard here.

**Comment by gustavo-iniguez-goya** [comment]
sorry, I meant: `$ sudo apt ./opensnitch* ./python3-opensnitch*` 

Bear in mind that in this case, the .deb packages must be on the same directory where you execute the apt command. Otherwise change `./` by the path to the deb packages.

anyway, a `sudo apt -f install` should resolve the dependencies.

**Comment by adim-att** [comment]
@gustavo-iniguez-goya 
I've used apt, it's an invalid operation. In Debian 11, my colleague installed via dpkg (Debian package manager). 
Since Ubuntu is Debian, I always install with dpkg -i *.deb. However, it shows dependency problems with Python3 libraries, most of them are for sql and notifications option with the system. It is my fault that in the question I tried `dpkg -i opensnitch*`, because I should have ended it with `.deb`, but still, there is a problem, as we now know.

apt install -f keeps installing an 800kb libs that don't fix the issue. 
I am kind of stuck. I think there is a loop here, one of the errors should mean something to us.

**Reply by gustavo-iniguez-goya** [reply]
oops, apparently you downloaded and installed 2 .deb packages of the GUI?

it looks like you downloaded python3-opensnitch-ui_1.6.0-rc.5-1_all.PIP.deb and python3-opensnitch-ui_1.6.0-rc.5-1_all.deb
On ubuntu 22.04 you only need python3-opensnitch-ui_1.6.0-rc.5-1_all.deb.

My suggestion would be remove python3-opensnitch-ui*, be sure that apt is in a clean state, then install the GUI:

`$ sudo apt install ./python3-opensnitch-ui_1.6.0-rc.5-1_all.deb`

**Reply by adim-att** [reply]
> oops, apparently you downloaded and installed 2 .deb packages of the GUI?
> 
> it looks like you downloaded python3-opensnitch-ui_1.6.0-rc.5-1_all.PIP.deb and python3-opensnitch-ui_1.6.0-rc.5-1_all.deb On ubuntu 22.04 you only need python3-opensnitch-ui_1.6.0-rc.5-1_all.deb.
> 
> My suggestion would be remove python3-opensnitch-ui*, be sure that apt is in a clean state, then install the GUI:
> 
> `$ sudo apt install ./python3-opensnitch-ui_1.6.0-rc.5-1_all.deb`

My apt is not in a clean state at all, it failed completely for some reason.
Thanks for your advisory, but I need to fix apt at some other time and get back to this.. will report back.

**Reply by adim-att** [reply]
> oops, apparently you downloaded and installed 2 .deb packages of the GUI?
> 
> it looks like you downloaded python3-opensnitch-ui_1.6.0-rc.5-1_all.PIP.deb and python3-opensnitch-ui_1.6.0-rc.5-1_all.deb On ubuntu 22.04 you only need python3-opensnitch-ui_1.6.0-rc.5-1_all.deb.
> 
> My suggestion would be remove python3-opensnitch-ui*, be sure that apt is in a clean state, then install the GUI:
> 
> `$ sudo apt install ./python3-opensnitch-ui_1.6.0-rc.5-1_all.deb`

Our company have migrated to _**Debian Testing Branch**_, we installed _OpenSnitch_ there and everything went fine. The question remains as to why _Ubuntu_ is undergoing these issues with _OpenSnitch_, maybe someone else, at one day will answer that.

Thank you !

**Comment by enlightdelightbroadway** [comment]
Same error happening on fedora atomic, do I have to file a new issue?
